### PR TITLE
JBPM-8232: Horizontal scrollbars missing in task inbox

### PR DESCRIPTION
**Pull request by nmirasch**
Added the ability (enableDataGridMinWidth) to configure the PagedTable to set a minimal width to ensure that have enough space to display the visible columns. In case that the required size is bigger than the avaliable space an horizonal scroll appears. The default column size is also configurable and it's applied when the column has not one fixed.

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz could you take a look please? Thanks!

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/ColumnPickerTest.java [hunk 1]
```diff
@@ -297,4 +297,43 @@ public void testAddColumnsIncrementally() {
                                         eq(100.0),
                                         eq(Style.Unit.PCT));
     }
+
+    @Test
+    public void testDataGridMinWidthCalculation() {
+        int defaultColumnWidthSize = 150;
+        int fixedColumn1Width = 38;
+
+        final Column column1 = createColumn("col1",
```
Comment (cristianonicolai) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/ColumnPickerTest.java]: similar in here, just avoid the extra break line. :)

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java [hunk 2]
```diff
@@ -114,4 +115,62 @@ public void testDefaultPageSizeValue() throws Exception {
         verify(select).setValue(String.valueOf(DEFAULT_PAGE_SIZE));
         verify(select).addValueChangeHandler(any());
     }
+
+    @Test
+    public void testDataGridMinWidthNotSetByDefault() throws Exception {
+        final int PAGE_SIZE = 10;
+        Element mockElement =mock(Element.class);
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java]: missing space around =

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java [hunk 3]
```diff
@@ -114,4 +115,62 @@ public void testDefaultPageSizeValue() throws Exception {
...
+        final int minWidth = 800;
+
+        PagedTable pagedTable = new PagedTable(PAGE_SIZE, null, null, false, false, false);
+        ColumnPicker columnPickerMock = mock(ColumnPicker.class);
+
+        when(pagedTable.dataGridContainer.getElement()).thenReturn(mockElement);
+        when(columnPickerMock.getDataGridMinWidth()).thenReturn(minWidth);
+        pagedTable.setColumnPicker(columnPickerMock);
+
+        pagedTable.enableDataGridMinWidth(true);;
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java]: extra semi-colon

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java [hunk 4]
```diff
@@ -114,4 +115,62 @@ public void testDefaultPageSizeValue() throws Exception {
...
+        final int minWidth = 800;
+
+        PagedTable pagedTable = new PagedTable(PAGE_SIZE, null, null, false, false, false);
+        ColumnPicker columnPickerMock = mock(ColumnPicker.class);
+
+        when(pagedTable.dataGridContainer.getElement()).thenReturn(mockElement);
+        when(columnPickerMock.getDataGridMinWidth()).thenReturn(minWidth);
+        pagedTable.setColumnPicker(columnPickerMock);
+        pagedTable.setTableHeight();
+        verify(mockElement,never()).setAttribute(eq("style"), anyString());
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java]: missing space after comma

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java [hunk 5]
```diff
@@ -114,4 +115,62 @@ public void testDefaultPageSizeValue() throws Exception {
...
+        PagedTable pagedTable = new PagedTable(PAGE_SIZE, null, null, false, false, false);
+        ColumnPicker columnPickerMock = mock(ColumnPicker.class);
+
+        when(pagedTable.dataGridContainer.getElement()).thenReturn(mockElement);
+        when(columnPickerMock.getDataGridMinWidth()).thenReturn(minWidth);
+        pagedTable.setColumnPicker(columnPickerMock);
+
+        pagedTable.enableDataGridMinWidth(true);;
+        pagedTable.setTableHeight();
+        verify(columnPickerMock,never()).setDefaultColumnWidthSize(anyInt());
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java]: missing space after comma

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java [hunk 6]
```diff
@@ -114,4 +115,62 @@ public void testDefaultPageSizeValue() throws Exception {
...
+        verify(mockElement).setAttribute("style", "min-width:" + minWidth + Style.Unit.PX.getType());
+
+    }
+
+    @Test
+    public void testDefaultColumnWidth() throws Exception {
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/test/java/org/uberfire/ext/widgets/common/client/tables/PagedTableTest.java]: formatting issues as in previous method

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java [hunk 7]
```diff
@@ -284,6 +284,12 @@ protected void resetTableColumns(int left,
     }
 
     protected void loadGlobalGridPreferences() {
+    }
+
+    public int getDataGridMinWidth(){
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java]: missing space

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java [hunk 8]
```diff
@@ -284,6 +284,12 @@ protected void resetTableColumns(int left,
...
+        return -1;
+    }
 
+    public void setDefaultColumnWidthSize(int defaultColumSize){
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java]: missing space

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.ui.xml [hunk 9]
```diff
@@ -58,9 +58,13 @@
       padding-top: 5px;
     }
 
+    .dataGridContainer{
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.ui.xml]: missing space

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java [hunk 10]
```diff
@@ -218,6 +237,10 @@ public HasWidgets getTopToolbar() {
         return topToolbar;
     }
 
+    protected void setColumnPicker(ColumnPicker columnPicker){
```
Comment (jstastny-cz) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java]: missing space

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz Thanks,
Added some code styling changes..

**Comment by nmirasch** [comment]
@jstastny-cz added some code style changes

**Pull request merged**